pkg/metrics/timer.go

```diff
@@ -0,0 +1,60 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Timer is the struct that keeps track of each of the sections.
+type Timer struct {
+	startTime    time.Time
+	totalTime    time.Time
+	listOfStages []string
+	stageTimes   map[string]time.Duration
+}
+
+const (
+	// InfrastructureStageName is a constant string value that is used to display the Infrastructure created stage text in the logs.
+	InfrastructureStageName = "Infrastructure"
+	// APIStageName is a constant string value that is used to display the API stage text in the logs.
+	APIStageName = "API"
+	// BootstrapCompleteStageName is a constant string value that is used to display the Bootstrap Complete text in the logs.
+	BootstrapCompleteStageName = "Bootstrap Complete"
+	// BootstrapDestroyStageName is a constant string value that is used to display the Boostrap Destroy Complete text in the logs.
+	BootstrapDestroyStageName = "Bootstrap Destroy"
+	// ClusterOperatorStageName is a constant string value that is used to display the Cluster Operator Complete text in the logs.
+	ClusterOperatorStageName = "Cluster Operator"
+
+	// ConsoleStageName is a constant string value that is used to display the Cluster Operator Complete text in the logs.
+	ConsoleStageName = "Console"
+)
+
+var timer Timer
+
+// StartTimer initiailzes the timer object with the current timestamp information.
+func StartTimer() {
+	timer = Timer{}
+	timer.startTime = time.Now().Round(time.Second)
+	timer.totalTime = time.Now().Round(time.Second)
+	timer.listOfStages = []string{}
+	timer.stageTimes = make(map[string]time.Duration)
+}
+
+// SetTimeElapsed returns the duration of the current stage.
+func SetTimeElapsed(key string) {
+	duration := time.Now().Round(time.Second).Sub(timer.startTime)
+	timer.listOfStages = append(timer.listOfStages, key)
+	timer.stageTimes[key] = duration
+	timer.startTime = time.Now().Round(time.Second)
+}
+
+// LogSummary prints the summary of all the times collected so far into the INFO section.
+func LogSummary() {
+	logrus.Debugf("Time elapsed for each section")
+	indent := "	"
+	for _, item := range timer.listOfStages {
+		logrus.Debugf(indent+"%s	:	%v", item, timer.stageTimes[item])
```

Comment:
I think there may be some tabs in here which are showing up when printed in CI. See: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/3263/pull-ci-openshift-installer-master-e2e-aws/10586/artifacts/e2e-aws/installer/.openshift_install.log

Use spaces instead of tabs.